### ISSUE TITLE
Warn about #[macro_export] mod m?

### ISSUE THREAD

**Issue by ralfbiedert** [rust-lang/rust#52371]
I haven't used macros that often, but had to make changes in a 3rd party project. 

During refactoring I tried to import macros from another module. I knew I had to add `macro_use`, `macro_export` somehow, but couldn't remember details.

While `rustc` was somewhat helpful in warning me on my first attempt:

```
error: cannot find macro `test!` in this scope
 --> src/x/u/u.rs:5:1
  |
5 | test! {
  | ^^^^^^^^^^
  |
  = help: have you added the `#[macro_use]` on the module/import?

```

It did not warn me about doing this:

```
#[macro_export] 
mod m;
```

where in reality I should have done that:

```
#[macro_use] 
mod m;
```

This was particularly confusing since the used macro was a few modules down in the hierarchy, and I should have done `#[macro_use]` on each level. Since I wasn't warned, I assumed `#[macro_export] mod m;` was correct. Instead, I tried a few permutations adding `#[macro_use]` to other places,  which all didn't work out. 

Two things that would have worked for me:

- Warn about `#[macro_export] mod m;`
- Make the help text above more explicit, e.g., by suggesting that `macro_use` needs to be added to all parent modules (maybe with a link to the [error index](https://doc.rust-lang.org/error-index.html))

**Comment by fmease** [comment]
Triage:

1. `#[macro_export]` on `mod m {}` / `mod m;` now leads to lint `unused_attributes` getting emitted (it can't be a hard-error (yet) for backward compatibility).
2. However, we still don't recommend `#[macro_use]` when encountering that user error.
